src/erlcloud_sts.erl

```diff
@@ -17,6 +18,18 @@
 -define(UTC_TO_GREGORIAN, 62167219200).
 -define(EXTERNAL_ID_MAX_LEN, 1224).
 
+% See http://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html
+% Options version of the API allowing MFA and etc access
+%% {"RoleArn", RoleArn},
```

Comment:
I like the idea of a /2 arity of `assume_role` that takes options, but I think I'd prefer to pass those as underscore-case atoms (`role_arn`, `role_session_name`, etc.) to be symmetric to the response format. Seems like maybe the `assume_role/2` could extract the options and call `assume_role/7` much as @kkuzmin suggested.